### ISSUE TITLE
Building a custom Turkish unitselection voice by replacing dfki-ot-data with my own recordings

### ISSUE THREAD

**Issue by elnursamedzade** [marytts/voice-dfki-ot#3]
Yes, I finally managed to compile the voice-dfki-ot voice with the ubuntu terminal I installed on windows.
Now, I want to replace the data in the build directory with my own data and recompile it. However, when I run gradlew build, the original data is downloaded again. How do I prevent this and add my own custom files to the build.gradle file?

**Comment by elnursamedzade** [comment]
Please come back to me. I need your help a lot.

**Comment by psibre** [comment]
@elnursamedzade Please see if you can build from the source branch of #4.

**Comment by elnursamedzade** [comment]
Yes, build is successful!
Will it work for me if I now replace my own wav and text files with those in the build directory?

**Comment by elnursamedzade** [comment]
I tried, but unfortunately it did not. I placed my own wav lab and txt files in the build directory, but again the original files were loaded.

**Comment by psibre** [comment]
The key is this part of the build configuration:

https://github.com/marytts/voice-dfki-ot/blob/483f374c0288c9aa4857c28f86aadf6920dd2190/build.gradle#L58-L64

This configures the source directory for the `wav` task (which is configured as an `InputDirectory` for the `basenames` task by default), and also configures the wav and lab input directories for the `basenames` task. For Gradle, this makes sure that the dfki-ot-data is downloaded and unpacked into `build/`, and if it is not present and up to date, then it will repeat those tasks, which are wired up as task dependencies.

If you want to replace the source data with your own, then you need to modify those assignments to use your own corresponding directories instead (see https://github.com/marytts/gradle-marytts-voicebuilding-plugin/issues/127#issuecomment-633389650 for an example).

See also the plugin documentation; from https://github.com/marytts/gradle-marytts-voicebuilding-plugin#project-layout:
> The audio location can be easily customized, by configuring the layout in the build.gradle like this:
> ```
> // wav files under "$projectDir/wav"
> wav.srcDir = file('wav')
> ```

**Comment by elnursamedzade** [comment]
Yes! I finally succeeded!
I realized that I had overlooked some minor nuances. How many months have I been having nightmares? Finally I wanted to!
Mr. Psibre, you're amazing! I will not forget you. Thank you very much!

**Comment by psibre** [comment]
Glad to hear it finally worked!

There will most likely be further issues related to the data itself. Please feel free to reach out with any questions.

**Comment by elnursamedzade** [comment]
Yes, I always got an error because of 2 lines of code. Finally, thanks to you, I succeeded. I love you!

**Comment by psibre** [comment]
@elnursamedzade Please have a look at [these slides](http://www.coli.uni-saarland.de/~steiner/teaching/2018/summer/voicebuilding/slides/) from a seminar on building custom voices for MaryTTS. I have a feeling it might clear up a few points.

**Comment by elnursamedzade** [comment]
Hello, Mr. Psibre.
Unfortunately, I cannot watch the slide because I am visually impaired.
The main problem is that when I replace wav files with my own wav files, I get the following error:
```
> Task :wav FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Failed to capture fingerprint of input files for task ':wav' property 'srcDir' during up-to-date check.
> Failed to create MD5 hash for file '/home/elka/voice-dfki-ot-upgrade-voicebuilding-plugin/wav/myvoice0001.wav' as it d
oes not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with
--scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 11s
3 actionable tasks: 3 executed
```
My `build.gradle` file:
```
plugins {
    id 'de.dfki.mary.voicebuilding-legacy' version '5.4'
}

marytts {
    voice {
        name = 'myvoice'
        gender = 'female'
        language = 'en'
        region = 'US'
        domain = 'general'
        type = 'unit selection'
        description = 'A female English unit selection voice'
        samplingRate = 16000
        license {
            name = 'Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International'
            shortName = 'CC BY-NC-SA 4.0'
            url = 'http://creativecommons.org/licenses/by-nc-sa/4.0/'
        }
    }
}

wav.srcDir = file('wav')
```
What am I doing wrong? Please enlighten me.

**Comment by psibre** [comment]
The source data must include *matching* audio (wav), text (orthographic), and lab (phonetic alignment) files for each utterance. Could you please double check that for each utterance (e.g., `myvoice0001`), you have corresponding files (`myvoice0001.wav`, `myvoice0001.txt`, and `myvoice0001.lab`) in the correct directories?

I'm confused about the "MD5 has" part of the error message though. Just to be safe, please remove your `build` directory to ensure that you start with a clean state. (Make sure you have your source data else where, of course!)

**Comment by elnursamedzade** [comment]
I am sending you my example work. Please check the file and let me know what I did wrong.
[](https://drive.google.com/file/d/1If8fqDGc_GDF9sj6NF5x_1cNT58G60ku/view?usp=sharing)

**Comment by psibre** [comment]
Not sure how you are "sending" your data. Assuming you're willing to share it, it would make sense if you created your own GItHub repository (using, e.g., this one as a template) and uploaded your data to a file sharing host. All of this can be private if you prefer.
